persistence-api/src/main/java/io/stargate/db/query/builder/Bindable.java

```diff
@@ -0,0 +1,34 @@
+package io.stargate.db.query.builder;
+
+import io.stargate.db.query.BindMarker;
+import io.stargate.db.query.BoundQuery;
+import io.stargate.db.query.TypedValue;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+
+public interface Bindable<B extends BoundQuery> {
+
+  TypedValue.Codec valueCodec();
+
+  /** The (potentially empty) list of bind markers that remains to be bound in this query. */
+  List<BindMarker> bindMarkers();
+
+  B bindValues(List<TypedValue> values);
```

Comment:
Since binding always produces a `BoundQuery`, what is the rationale for splitting `Bindable` from `Query` (which is its only sub-interface)? Do we expect other, non-query implementations of `Bindable`?

Reply:
I think it's just a remnant of my iterations developing this. I indeed don't expect other implementations. I'll fold it back into `Query` directly. Thanks for noticing.